test/Exception.js

```diff
@@ -191,19 +210,29 @@ describe('Exception', () => {
 		})
 	})
 
-	it('should recover from error boundary in the event of an exception', () => {
-		const target = document.createElement('div')
-		const Primary = props => {
-			const [error, dispatch] = useState('')
+	it('should async catch exceptions', (done) => {
+	  const target = document.createElement('div')
+		const stack = assert.spyr(console, 'error')
+	  const Primary = props => {
+	    return h('div', {}, Promise.reject('x'))
+	  }
 
-			return h(Boundary, {fallback: error => dispatch(error.message)}, error ? error : h(Secondary))
-		}
-		const Secondary = props => {
-			throw 1
-		}
+	  render(h(Primary), target).then(null, (current) => {
+	    done(assert.equal(current, 'x'))
+	  })
+	})
 
-		render(h('div', {}, h(Primary)), target, (current) => {
-			assert.html(current, '<div>1</div>')
-		})
+	it('should not async catch exceptions', (done) => {
```

Comment:
not sure I understand the purpose of this test

Reply:
There are no exceptions raised in resolution of the promise so the second callback to `.then()` should not be executed. That is what this is testing – false positives.

Reply:
Perhaps the title should be `should not async catch non-exceptions`
I'm wondering if the result of `render()` should instead be an `instanceof Promise`, to ensure Promises/A+ spec compliance.